docs/FAQ.rst

```diff
@@ -0,0 +1,276 @@
+
+Q: How to debug the DALI pipeline?
+##################################
+A: Just recently DALI added `an eager debug mode <../examples/general/debug_mode.html>`_ so
+the output of each operator can be instantly evaluated and andy python code can be added inside
+the pipeline for the prototyping purpose.
+
+Q: Can I access the contents of intermediate data nodes in the pipeline?
+########################################################################
+A: In the pipeline mode it is not possible, however, thanks to the recently introduced
+`debug mode <../examples/general/debug_mode.html>`_` it can be done for prototyping
+and debugging purposes.
+
+Q: When will DALI support the XYZ operator?
+###########################################
+A: We cannot commit to any timeline to add any particular operator, still we are open to external
+contributions. On top of that every user can extend DALI on his own without modifying its code
+base, please check `this page <../examples/custom_operations/index.html>`_ for more details.
+
+Q: How should I know if I should use a CPU or GPU operator variant?
+###################################################################
+A: It depends on the particular use case. I would start by checking GPU  utilization to see if
+throwing more work on the GPU won't slow things down, or just place everything on the CPU first
+and then gradually move things to the GPU and see if that helps. You can check
+`our GTC22 talk <https://www.nvidia.com/gtc/session-catalog/#/session/1636559250287001p4DG>`_
+or `the blog post <https://developer.nvidia.com/blog/case-study-resnet50-dali/>`_ to see how
+to do it step by step.
+
+Q: How can I provide a custom data source/reading pattern to DALI?
+##################################################################
+A: You can define any custom data loading pattern using python and
+`external source operator <../examples/general/data_loading/external_input.html>`_. To make it
+faster please use `its parallel capability <../examples/general/data_loading/parallel_external_source.html>`_.
+
+Q: Does DALI provide any profiling capabilities?
+################################################
+A: DALI doesn't provide any built-in profiling capabilities, still it utilizes NVTX ranges
+and has a dedicated domain (so it is easy to find in the profile) to show its operations. So you can
+capture the profile using `NVIDIA Nsight Systems <https://developer.nvidia.com/nsight-systems>`_
+or any Deep Learning profile that also supports NVTX markers.
+
+Q: Does DALI support multi GPU/node training?
+#############################################
+A: Yes, DALI supports data-parallel and distributed data-parallel strategies. Its shards data into
+non-overlapping pieces using the number of shards (world size) and shard id (global rank), and
+uses device id to identify the GPU used in the particular node (local rank).
```

Comment:
I don't quite understand it, so I doubt the readers unfamiliar with the product would.

Reply:
Reworked